# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/cairo/cairo/src/cairo-matrix.c
# Commit: 3cdfb39376b7
# Full Hash: 3cdfb39376b736597fe658a322389117686b7e82
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-05-07 09:13:07
# Regressor Bug: 1892913
# File Overlap Count: 3
# Description:
#   Bug 1892913 - patch 1 - Update cairo source to 1.18.0 r=gfx-reviewers,lsalzman
#   
#   Wholesale replacement of files in gfx/cairo/cairo with contents of the 1.18.0 release tarball,
#   omitting subdirectories that are not part of the core library build.
#   
# ==============================================================================

diff -r 9daa351f875b -r 3cdfb39376b7 gfx/cairo/cairo/src/cairo-matrix.c
--- a/gfx/cairo/cairo/src/cairo-matrix.c	Mon May 06 19:22:35 2024 +0000
+++ b/gfx/cairo/cairo/src/cairo-matrix.c	Mon May 06 19:25:12 2024 +0000
@@ -85,7 +85,6 @@
 		       0, 1,
 		       0, 0);
 }
-slim_hidden_def(cairo_matrix_init_identity);
 
 /**
  * cairo_matrix_init:
@@ -118,7 +117,6 @@
     matrix->xy = xy; matrix->yy = yy;
     matrix->x0 = x0; matrix->y0 = y0;
 }
-slim_hidden_def(cairo_matrix_init);
 
 /**
  * _cairo_matrix_get_affine:
@@ -178,7 +176,6 @@
 		       0, 1,
 		       tx, ty);
 }
-slim_hidden_def(cairo_matrix_init_translate);
 
 /**
  * cairo_matrix_translate:
@@ -202,7 +199,6 @@
 
     cairo_matrix_multiply (matrix, &tmp, matrix);
 }
-slim_hidden_def (cairo_matrix_translate);
 
 /**
  * cairo_matrix_init_scale:
@@ -224,7 +220,6 @@
 		       0, sy,
 		       0, 0);
 }
-slim_hidden_def(cairo_matrix_init_scale);
 
 /**
  * cairo_matrix_scale:
@@ -247,7 +242,6 @@
 
     cairo_matrix_multiply (matrix, &tmp, matrix);
 }
-slim_hidden_def(cairo_matrix_scale);
 
 /**
  * cairo_matrix_init_rotate:
@@ -277,7 +271,6 @@
 		       -s, c,
 		       0, 0);
 }
-slim_hidden_def(cairo_matrix_init_rotate);
 
 /**
  * cairo_matrix_rotate:
@@ -343,7 +336,6 @@
 
     *result = r;
 }
-slim_hidden_def(cairo_matrix_multiply);
 
 void
 _cairo_matrix_multiply (cairo_matrix_t *r,
@@ -372,15 +364,10 @@
  * the returned vector is as follows:
  *
  * <programlisting>
- * dx2 = dx1 * a + dy1 * c;
- * dy2 = dx1 * b + dy1 * d;
+ * dx_new = xx * dx + xy * dy;
+ * dy_new = yx * dx + yy * dy;
  * </programlisting>
  *
- * Affine transformations are position invariant, so the same vector
- * always transforms to the same vector. If (@x1,@y1) transforms
- * to (@x2,@y2) then (@x1+@dx1,@y1+@dy1) will transform to
- * (@x1+@dx2,@y1+@dy2) for all values of @x1 and @x2.
- *
  * Since: 1.0
  **/
 void
@@ -394,7 +381,6 @@
     *dx = new_x;
     *dy = new_y;
 }
-slim_hidden_def(cairo_matrix_transform_distance);
 
 /**
  * cairo_matrix_transform_point:
@@ -414,7 +400,6 @@
     *x += matrix->x0;
     *y += matrix->y0;
 }
-slim_hidden_def(cairo_matrix_transform_point);
 
 void
 _cairo_matrix_transform_bounding_box (const cairo_matrix_t *matrix,
@@ -625,7 +610,6 @@
 
     return CAIRO_STATUS_SUCCESS;
 }
-slim_hidden_def(cairo_matrix_invert);
 
 cairo_bool_t
 _cairo_matrix_is_invertible (const cairo_matrix_t *matrix)